File: src/ArticlesApp.WebAPI/AppFrontend/src/Entities/Notification.ts
import { NotificationTypesEnum } from "src/Entities/NotificationTypesEnum";



export class Notification {
  constructor(notif: Notification) {
    this.id = notif.id;
    this.message = notif.message;
    this.reciever_UserId = notif.reciever_UserId;
    this.createdAt_DateUtc = notif.createdAt_DateUtc;
    // this.readAt_DateUtc = notif.readAt_DateUtc;
    // this.referencedArticle_ArticleId = notif.referencedArticle_ArticleId;
    // this.notificationType_TypeId = notif.notificationType_TypeId;
  }

  id: number;

  message: string;

  reciever_UserId: string;

  createdAt_DateUtc: Date;

  readAt_DateUtc?: Date;

  referencedArticle_ArticleId?: number;

  notificationType_TypeId?: NotificationTypesEnum;
}

File: src/ArticlesApp.Core/Entities/Notification/Notification.cs
namespace ArticlesApp.Core.Entities.Notification;



public class Notification : INotification
{
    public Notification(
        int id,
        string message,
        string reciever_UserId,
        DateTime createdAt_DateUtc
    )
    {
        Id = id;
        Message = message;
        Reciever_UserId = reciever_UserId;
        CreatedAt_DateUtc = createdAt_DateUtc;
    }

    public int Id { get; set; }

    public string Message { get; set; }

    public string Reciever_UserId { get; set; }

    public DateTime CreatedAt_DateUtc { get; set; }

    public DateTime? ReadAt_DateUtc { get; set; }

    public NotificationTypesEnum? NotificationType_TypeId { get; set; }

    public int? ReferencedArticle_ArticleId { get; set; }
}